lib/rouge/lexers/solidity.rb

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Solidity < RegexLexer
+      title "Solidity"
+      desc "Solidity, an Ethereum smart contract programming language"
+      tag 'solidity'
+      filenames '*.sol', '*.solidity'
+      mimetypes 'text/x-solidity'
+
+      # optional comment or whitespace
+      ws = %r((?:\s|//.*?\n|/[*].*?[*]/)+)
+      id = /[a-zA-Z$_][\w$_]*/
+
+      def self.detect?(text)
+        return true if text.start_with? 'pragma solidity'
+      end
+
+      # TODO: seperate by "type"
+      def self.keywords
+        @keywords ||= Set.new %w(
+          abstract anonymous as assembly break catch constant constructor continue
+          contract do delete else emit enum event external fallback for function hex
+          if indexed interface internal import is library mapping memory
+          modifier new payable public pure pragma private return returns
```

Comment:
`receive`, `override`, `virtual` are missing. They were added in 0.6.0.

Reply:
Ah, they're still stuck in `reserved`. A moment.